artifact/local/fs.go

```diff
@@ -0,0 +1,89 @@
+package local
+
+import (
+	"context"
+	"crypto/sha256"
+	"encoding/json"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	digest "github.com/opencontainers/go-digest"
+	"golang.org/x/xerrors"
+
+	"github.com/aquasecurity/fanal/analyzer"
+	_ "github.com/aquasecurity/fanal/analyzer/os/alpine"
+	_ "github.com/aquasecurity/fanal/analyzer/pkg/apk"
+	"github.com/aquasecurity/fanal/artifact"
+	"github.com/aquasecurity/fanal/cache"
+	"github.com/aquasecurity/fanal/types"
+	"github.com/aquasecurity/fanal/walker"
+)
+
+type Artifact struct {
+	dir   string
+	cache cache.ArtifactCache
+}
+
+func NewArtifact(dir string, c cache.ArtifactCache) artifact.Artifact {
+	return Artifact{
+		dir:   dir,
+		cache: c,
+	}
+}
+
+func (a Artifact) Inspect(_ context.Context) (types.ArtifactReference, error) {
+	var result analyzer.AnalysisResult
+	err := walker.WalkDir(a.dir, func(filePath string, info os.FileInfo, opener analyzer.Opener) error {
+		filePath, err := filepath.Rel(a.dir, filePath)
+		if err != nil {
+			return err
+		}
+		r, err := analyzer.AnalyzeFile(filePath, info, opener)
+		if err != nil {
+			return err
+		}
+		result.Merge(r)
+		return nil
+	})
+	if err != nil {
+		return types.ArtifactReference{}, err
+	}
+
+	blobInfo := types.BlobInfo{
+		SchemaVersion: types.BlobJSONSchemaVersion,
+		OS:            result.OS,
+		PackageInfos:  result.PackageInfos,
+		Applications:  result.Applications,
+	}
+
+	// calculate hash of JSON and use it as pseudo artifactID and blobID
+	h := sha256.New()
+	if err = json.NewEncoder(h).Encode(blobInfo); err != nil {
+		return types.ArtifactReference{}, err
+	}
+
+	d := digest.NewDigest(digest.SHA256, h)
+	diffID := d.String()
+	blobInfo.DiffID = diffID
+
+	if err = a.cache.PutBlob(diffID, blobInfo); err != nil {
+		return types.ArtifactReference{}, xerrors.Errorf("failed to store blob (%s) in cache: %w", diffID, err)
+	}
+
+	// get hostname
+	var hostName string
+	b, err := ioutil.ReadFile(filepath.Join(a.dir, "etc", "hostname"))
+	if err != nil {
+		hostName = a.dir
+	} else {
+		hostName = strings.TrimSpace(string(b))
+	}
```

Comment:
What's this for?

Reply:
if you scan filesystem inside a container like `trivy filesystem /`, it shows `/ (alpine:3.11)`. I thought we should display the hostname. e.g. `web01 (alpine:3.11)`.

Reply:
I see. How about using `os.Hostname` in that case?

Reply:
I didn't know it. I'll use it. Thanks!

Reply:
Oh, I'm sorry, but I found `os.Hostname` doesn't work in our case. When a user scans `/app/python_project`, we shouldn't show the hostname of the local machine. They focus on scanning the project and don't want to know the hostname.

Also, they may export a container's filesystem and scan the directory. In that case, we should display the hostname of the container, not the hostname of the machine running the container.

For example,
```
$ hostname
knqyf263's Mac
$ docker export $(docker create alpine:3.7) |tar -C /tmp/rootfs -xvf -
$ ls /tmp/rootfs
bin/  dev/  etc/  home/  lib/  media/  mnt/  proc/  root/  run/  sbin/  srv/  sys/  tmp/  usr/  var/
$ trivy fs /tmp/rootfs
```

In this case, we shouldn't display `knqyf263's Mac`, but the content of `/tmp/rootfs/etc/hostname`. If we use `os.Hostname()`, it returns `knqyf263's Mac`.